lab-davide/test/task-router-test.js

```diff
@@ -0,0 +1,174 @@
+
+'use strict';
+
+process.env.MONGO_URI = 'mongodb://localhost/test';
+
+const debug = require('debug')('task:task-router-test');
+const expect = require('chai').expect;
+const request = require('superagent-use');
+const superPromise = require('superagent-promise-plugin');
+
+request.use(superPromise);
+
+//app modules/
+const server = require('../server');
+const taskCrud = require('../lib/task-crud');
+
+//required modules dependent on global env//
+const port = process.env.PORT || 3000;
+const baseUrl = `http://localhost:${port}/api/task`;
+
+
+
+describe('testing module task-router', function() {
+  debug('hitting module task router');
+  before((done) => {
+    if (!server.isRunning) {
+      server.listen(port, () => {
+        server.isRunning = true;
+        debug('server running on port', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if(server.isRunning) {
+      debug('server is close');
+      server.close(() => {
+        server.isRunning = false;
+        console.log('server down');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing GET /api/task/:id with a valid id', function() {
+    before((done) => {
+      taskCrud.createTask({content: 'test task', desc: 'test data', dueDate: '2016-06-21'})
+      .then(task => {
+        debug('created task for GET tests!!!');
+        this.tempTask = task;
+        console.log('task task', this.tempTask);
+        done();
+      })
+      .catch(() => {
+        debug('Failed to create task for GET tests');
+        done();
+      });
+    });
+
+    it('should return a task', (done) => {
+      debug('GET task route: '+`${baseUrl}/${this.tempTask._id}`);
+      request.get(`${baseUrl}/${this.tempTask._id}`)
+      .then((res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.desc).to.equal('test data');
+        done();
+      })
+      .catch(() => {
+        expect(true).to.equal(false);
+        done();
+      });
+    });
+
+    it('should return not found', (done) => {
+      debug('GET task route not found: '+`${baseUrl}/${this.tempTask._id}`);
+      request.get(`${baseUrl}/123${this.tempTask._id}`)
+      .then(done)
+      .catch((err) => {
+        try {
+          var res = err.response;
+          expect(res.status).to.equal(404);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+  });
+
+  describe('POST /api/task with valid data', function (){
+    debug('post api/task is working?');
+    before((done) =>
+      taskCrud.createTask({content: 'test task', desc: 'test data', dueDate: '2016-06-21'})
```

Comment:
That string can be coerced into a valid date object, but you may just want to use something like Date.now() or new Date() in the future to avoid formatting issues.